tests/extrusion/test_point_eval_fs_extrusion.py

```diff
@@ -0,0 +1,132 @@
+def test_quad(mesh_quad, family, degree, vfamily, vdegree):
+    V = FunctionSpace(mesh_quad, family, degree, vfamily=vfamily, vdegree=vdegree)
+    f = Function(V).interpolate(Expression("(x[0] - 0.5)*(x[0] - x[1])"))
+    assert np.allclose(0.02, f([0.6, 0.4]))
+    assert np.allclose(0.45, f([0.0, 0.9]))
+
+
+@pytest.mark.parametrize(('family', 'degree'),
+                         [('CG', 1),
+                          ('DG', 1),
+                          ('RTCF', 2),
+                          ('RTCE', 2)])
+def test_quad_vector(mesh_quad, family, degree):
+    if family in ['CG', 'DG']:
+        V = VectorFunctionSpace(mesh_quad, family, degree)
+        f = Function(V).interpolate(Expression(("0.2 + x[1]", "0.8*x[0] + 0.2*x[1]")))
+    else:
+        V = FunctionSpace(mesh_quad, family, degree)
+        f = Function(V).project(Expression(("0.2 + x[1]", "0.8*x[0] + 0.2*x[1]")))
+
+    assert np.allclose([0.6, 0.56], f([0.6, 0.4]))
+    assert np.allclose([1.1, 0.18], f([0.0, 0.9]))
+
+
+@pytest.mark.parametrize(('family', 'degree', 'vfamily', 'vdegree'),
+                         [('CG', 3, 'DG', 2),
+                          ('DG', 3, 'CG', 2)])
+def test_prism(mesh_prism, family, degree, vfamily, vdegree):
+    V = FunctionSpace(mesh_prism, family, degree, vfamily=vfamily, vdegree=vdegree)
+    f = Function(V).interpolate(Expression("(x[0] - 0.5)*(x[1] - x[2])"))
+    assert np.allclose(0.01, f([0.6, 0.4, 0.3]))
+    assert np.allclose(0.06, f([0.2, 0.6, 0.8]))
+
+
+@pytest.mark.parametrize('args',
+                         [('CG', 1),
+                          ('DG', 1),
+                          EnrichedElement(
+                              HDiv(OuterProductElement(
+                                  FiniteElement('RT', triangle, 2),
+                                  FiniteElement('DG', interval, 1))),
+                              HDiv(OuterProductElement(
+                                  FiniteElement('DG', triangle, 1),
+                                  FiniteElement('CG', interval, 2)))),
+                          EnrichedElement(
+                              HCurl(OuterProductElement(
+                                  FiniteElement('RT', triangle, 2),
+                                  FiniteElement('CG', interval, 2))),
+                              HCurl(OuterProductElement(
+                                  FiniteElement('CG', triangle, 2),
+                                  FiniteElement('DG', interval, 1))))])
+def test_prism_vector(mesh_prism, args):
+    if isinstance(args, tuple):
+        V = VectorFunctionSpace(mesh_prism, *args)
+        f = Function(V).interpolate(Expression(("0.2 + x[1]", "0.8*x[0] + 0.2*x[2]", "x[1]")))
+    else:
+        V = FunctionSpace(mesh_prism, args)
+        f = Function(V).project(Expression(("0.2 + x[1]", "0.8*x[0] + 0.2*x[2]", "x[1]")))
+
+    assert np.allclose([0.6, 0.54, 0.4], f([0.6, 0.4, 0.3]))
+    assert np.allclose([0.8, 0.32, 0.6], f([0.2, 0.6, 0.8]))
```

Comment:
Disgusting ugliness, but to avoid hardcoding the result, you could have done something like:

```
expr = ("0.2 + x[1]", "0.8*x[0] + 0.2*x[2]", "x[1]")
f = unction(V).project(Expression(expr))

x = [0.6, 0.4, 0.3]

actual = f(x)

expect = map(eval, expr)

assert np.allclose(expect, actual)
```